src/Microsoft.CodeQuality.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -0,0 +1,357 @@
+        }
+
+        /// <summary>
+        /// Checks whether the value of the invocation of <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/>
+        /// is being compared with 0 using <see cref="int.Equals(int)"/>.
+        /// </summary>
+        /// <param name="invocationOperation">The invocation operation.</param>
+        /// <param name="enumerableType">Type of the enumerable.</param>
+        /// <returns><see langword="true" /> if the value of the invocation of <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/>
+        /// is being compared with 0 using <see cref="int.Equals(int)"/>; otherwise, <see langword="false" />.</returns>
+        private static bool IsCountEqualsZero(IInvocationOperation invocationOperation, INamedTypeSymbol enumerableType)
+        {
+            if (!TryGetInt32Constant(invocationOperation.Arguments[0].Value, out var constant) || constant != 0)
+            {
+                return false;
+            }
+
+            return IsCountMethodInvocation(invocationOperation.Instance, enumerableType);
+        }
+
+        /// <summary>
+        /// Checks whether 0 is being compared with the value of the invocation of 
+        /// <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/>
+        /// using <see cref="int.Equals(int)"/>.
+        /// </summary>
+        /// <param name="invocationOperation">The invocation operation.</param>
+        /// <param name="enumerableType">Type of the enumerable.</param>
+        /// <returns><see langword="true" /> if 0 is being compared with the value of the invocation of 
+        /// <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/>
+        /// using <see cref="int.Equals(int)"/>; otherwise, <see langword="false" />.</returns>
+        private static bool IsZeroEqualsCount(IInvocationOperation invocationOperation, INamedTypeSymbol enumerableType)
+        {
+            if (!TryGetInt32Constant(invocationOperation.Instance, out var constant) || constant != 0)
+            {
+                return false;
+            }
+
+            return IsCountMethodInvocation(invocationOperation.Arguments[0].Value, enumerableType);
+        }
+
+        /// <summary>
+        /// Check to see if we have an expression comparing the result of
+        /// <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})"/> or
+        /// <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource}, Func{TSource, bool})"/>
+        /// using operators.
+        /// </summary>
+        private static void AnalyzeBinaryExpression(IBinaryOperation binaryOperation, INamedTypeSymbol enumerableType, Action<Diagnostic> reportDiagnostic)
+        {
+            if (binaryOperation.OperatorKind != BinaryOperatorKind.Equals &&
```

Comment:
Nit: Might be cleaner to move it to an extension method, say `AnyBinaryEqualsOperatorKind`, and use a switch case.